repondeur/zam_repondeur/services/fetch/http.py

```diff
@@ -1,16 +1,29 @@
-import os
-from pathlib import Path
+from typing import Optional
 
 import requests
 from cachecontrol import CacheControl
 from cachecontrol.caches.file_cache import FileCache
+from pyramid.config import Configurator
+from pyramid.registry import Registry
+from pyramid.threadlocal import get_current_registry
+from zope.interface import Interface
 
-HERE = Path(__file__)
-DEFAULT_HTTP_CACHE_DIR = HERE.parent.parent.parent / ".web_cache"
 
-HTTP_CACHE_DIR = os.environ.get("ZAM_HTTP_CACHE_DIR", DEFAULT_HTTP_CACHE_DIR)
+class IHTTPSession(Interface):
+    pass
 
 
-session = requests.session()
+def includeme(config: Configurator) -> None:
+    """
+    Called automatically via config.include("zam_repondeur.services.fetch.http")
+    """
+    session = requests.session()
+    http_cache_dir = config.registry.settings["zam.http_cache_dir"]
+    cached_session = CacheControl(session, cache=FileCache(http_cache_dir))
+    config.registry.registerUtility(component=cached_session, provided=IHTTPSession)
 
-cached_session = CacheControl(session, cache=FileCache(HTTP_CACHE_DIR))
+
+def get_http_session(registry: Optional[Registry] = None) -> CacheControl:
```

Comment:
Do we use that optional param somewhere?

Reply:
Not yet, but I feel less bad if I give the caller the opportunity of passing it as a param and avoiding a `get_current_registry()` call when not necessary...

(Similar to https://github.com/betagouv/zam/blob/master/repondeur/zam_repondeur/templating.py#L43-L45)

Reply:
Got it 👍